Command

shinsuke-nara opened the pull request:
I designed new API to post new command with title, description and meta data.

Could you review this?

This PR must not be merged.

Related issue No 788.

shinsuke-nara commented:
@satoshikumano Could you review this?

satoshikumano commented on thingif/src/main/java/com/kii/thingif/ThingIFAPI.java:
```diff
@@ -416,9 +417,11 @@ public void uninstallPush(@NonNull String installationID) throws ThingIFExceptio
      * of the command by calling {@link #getCommand}.
      * @throws ThingIFException Thrown when failed to connect IoT Cloud Server.
      * @throws ThingIFRestException Thrown when server returns error response.
+     * @deprecated use {@link ThingIFAPI#postNewCommand(CommandForm)}.
```
Lets' not deprecated this method.
Still useful when Title/ Metadata/ Description is not required.

shinsuke-nara commented:
@satoshikumano https://github.com/KiiPlatform/thing-if-AndroidSDK/pull/75#discussion_r58328207
I will remove deprecated mark.

shinsuke-nara commented:
@satoshikumano I removed deprecated from `postNewCommand()`.

I also added getter of fields. They are required to get values in CommandFrom at ThingIFAPI class.
I think such getters is not required as public API, but to get these values, They are required to be public API.

If we can move CommandForm class to `com.kii.thingif` package, These getters can be package internal. I think it is better idea.

What do you think?

shinsuke-nara commented:
@naoki-akimoto Could you implement this feature? Please create new brunch and make PR to this brunch.

satoshikumano commented:
@shinsuke-nara LGTM

shinsuke-nara commented on thingif/src/main/java/com/kii/thingif/command/CommandForm.java:
```diff
@@ -0,0 +1,157 @@
+    /**
+     * Setter of title
+     *
+     * @param title Length of title must be equal or less than 50.
```
50 characters.

shinsuke-nara commented:
Spec review is finished.

nfukuzaki commented on thingif/src/main/java/com/kii/thingif/ThingIFAPI.java:
```diff
@@ -447,6 +448,57 @@ public Command postNewCommand(
+        Map<String, String> headers = this.newHeader();
+        Command command = new Command(schemaName, schemaVersion, this.owner.getTypedID(),
+                form.getActions());
+        if (form.getTitle() != null) {
```
Nullableなのでif文は不要な気がします。
description, metadataも同様です。

shinsuke-nara replied:
@nfukuzaki  commandのsetterがnullを受け付けられるので、if文は不要ですね。

@naoki-akimoto 対応お願いします。

naoki-akimoto commented:
@shinsuke-nara @nfukuzaki 
不要なif文の削除しました。

nfukuzaki commented:
LGTM
